icechunk-python/python/icechunk/store.py

```diff
@@ -65,26 +62,29 @@ def __eq__(self, value: object) -> bool:
 
     def __getstate__(self) -> object:
         # we serialize the Rust store as bytes
-        if not self._allow_pickling and not self._store.read_only:
-            raise ValueError(
-                "You must opt in to pickling this *writable* store by using `Session.allow_pickling` context manager"
-            )
         d = self.__dict__.copy()
         d["_store"] = self._store.as_bytes()
+        d["_for_fork"] = self._for_fork
         return d
 
     def __setstate__(self, state: Any) -> None:
         # we have to deserialize the bytes of the Rust store
         store_repr = state["_store"]
         state["_store"] = PyStore.from_bytes(store_repr)
-        state["_read_only"] = state["_store"].read_only
+        if not state["_for_fork"]:
+            state["_read_only"] = False
+        else:
+            state["_read_only"] = state["_store"].read_only
```

Comment:
This ugliness would fix this UX bug https://github.com/earth-mover/icechunk/issues/964